lib/ssl/src/ssl.erl

```diff
@@ -2128,11 +2129,12 @@ connect(Host, Port, TLSOptions, infinity).
 -spec connect(TCPSocketOrHost, TLSOptionsOrPort, TimeoutOrTLSOptions) ->
           {ok, sslsocket()} |
           {ok, sslsocket(), Ext :: protocol_extensions()} |
-          {error, reason()} |
-          {option_not_a_key_value_tuple, any()} when
+          {error, Reason} when
       TCPSocketOrHost :: socket() | host(),
       TLSOptionsOrPort :: [tls_client_option()] | inet:port_number(),
-      TimeoutOrTLSOptions :: [tls_client_option()] | timeout().
+      TimeoutOrTLSOptions :: [tls_client_option()] | timeout(),
+      Reason :: closed | timeout | {options, any()} |
+                error_alert() | reason().
```

Comment:
what is the value of being more precise in specifying possible error reasons and finally appending `reason()` which is `term()` ?
Is it only for the reader comfort?
why `term()`  (a very general type spec) is needed? is ssl receiving some of the reasons from other applications and they're unspecified?

Reply:
thanks for explanation.

> The reason() type was already exported otherwise I probably would have just put term() in the end.

yes. going to `reason()` declaration and finding it to be `term()` is disappointing.

Reply:
At least it also has an explanation saying you should not match it :)